### PR TITLE
Kalendereintraege erstellen

### LINKED ISSUE #27: creating appointments
At the beginning of a new semester, a VLA member clicks somewhere (tbd) in the calendar, and a page to create a new appointment opens. This form has fields to enter 
* the title
* the lecture type it belongs to (optional) (#34)
* category (lecture/vacation/Abnahmetermin/...) (#28)
* start datetime, end datetime, recurrence week days, recurrence end date,
* peoples / lecturers: there may be multiple people (Prof, Assistenz) per appointment (#33)

After submitting the form, this appointment is visible in the main calendar view, colored according to its category.

---

Questions to be answered:

* [x] person: list of existing people or free-text form?

ToDo:

* [x] lectures -> #69 
* [x] appointment categories -> #70 
* [x] persons -> #71 
* [ ] lecture_persons
* [x] appointment_series -> #99 
* [x] appointments -> #99 
* [x] acceptance -> #99 
* [x] experiment_booking -> #99

### PR DESCRIPTION
**Pull request by polodasi123**
<comment on what this PR does>

closes #27 

---

### Implementation

- [x] Does this code change accomplish what it is supposed to do?
- [x] Can this solution be simplified without unreasonable effort
- [x] Is the code modular enough?
- [x] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [x] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [x] Interface Segregation: If interface implemented, is it fully implemented?
- [x] Dependency Injection: Are central object instantiation processes managed by a single context?
- [x] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [ ] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [x] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [x] Is error handling done the correct way?
- [x] Should any logging or debugging information be added or removed?
- [x] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [x] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [x] Is data retrieved from external APIs or libraries checked for security issues?
- [x] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [x] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [x] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [x] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by smaller methods/different names?
- [x] Is the code located in the right file/folder?
- [x] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message better?
- [ ] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

File: VLA-Frontend/src/components/calendar/dateUtils.ts [hunk 1]
```diff
@@ -83,3 +83,20 @@ const rangeFmt = new Intl.DateTimeFormat("de-DE", {
 export function formatRangeShortDE(start: Date, end: Date) {
   return `${rangeFmt.format(start)} – ${rangeFmt.format(end)}`;
 }
+/** Parse yyyy-mm-dd as a *local* Date (prevents timezone shift). */
+export function parseISODateLocal(iso: string): Date {
+  const [datePart] = iso.split("T"); // in case you ever store timestamps
+  const [y, m, d] = datePart.split("-").map(Number);
+  return new Date(y, (m ?? 1) - 1, d ?? 1);
+}
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/components/calendar/dateUtils.ts]: I think we should avoid having to parse an ISO string to a local date.
We should remove this function and replace the data types accordingly.

File: VLA-Frontend/src/components/calendar/DayColumn.tsx [hunk 2]
```diff
@@ -1,22 +1,50 @@
 
-import type { CalendarDay } from "./CalendarTypes";
+import type { CalendarDay, CalendarEvent , EventStatus } from "./CalendarTypes";
 
 type DayColumnProps = {
   day: CalendarDay;
+  events: CalendarEvent[];
+  onEventClick?: (event: CalendarEvent) => void;
+  getEventColor?: (event: CalendarEvent) => string | undefined;
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/components/calendar/DayColumn.tsx]: Maybe narrow the string down to actual HTML colors.
I think that not providing a getEventColor should automatically yield undefinded as a color.
This makes the undefinded return type not have a purpose

File: VLA-Frontend/src/components/calendar/DayColumn.tsx [hunk 3]
```diff
@@ -1,22 +1,50 @@
...
+          key={event.id}
+          className={`cv-event cv-event-${event.kind} ${getStatusClass(
+            event.status
+          )}`}
+          style={customColor ? { 
+              backgroundColor: customColor,
+              borderColor: customColor 
+            } : undefined}
+            title={event.subtitle || event.title}
+            onClick={() => onEventClick && onEventClick(event)}
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/components/calendar/DayColumn.tsx]: Just remove onClick entirely when onEventClick is not defined

File: VLA-Frontend/src/styles/CalendarView.css [hunk 4]
```diff
@@ -83,12 +84,14 @@
...
   background: #d9d9d9;
+  overflow-y: auto;
 }
 
 .cv-dayColumn {
   border-right: 1px solid #777;
   padding: 10px;
   background: transparent;
+  min-height: 100%;
 }
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/styles/CalendarView.css]: Make the dayColumn scrollable, we want to avoid a scrolling website

File: VLA-Frontend/src/components/calendar/WeekdayPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  { value: 3, label: "Mi" },
+  { value: 4, label: "Do" },
+  { value: 5, label: "Fr" },
+];
+
+export default function WeekdayPicker({
+  selectedDays,
+  onDaysChange,
+}: WeekdayPickerProps) {
+  const toggleWeekday = (day: number) => {
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/components/calendar/WeekdayPicker.tsx]: Why are you declaring this as a arrow function when f(param) is used almost everywhere else

File: VLA-Frontend/src/hooks/useCalendarNavigation.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+} from "../components/calendar/dateUtils";
+
+/**
+ * useCalendarNavigation manages the currently displayed workday range.
+ * It supports:
+ * - previous/next workday navigation
+ * - jumping to a specific date
+ */
+
+export function useCalendarNavigation() {
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/hooks/useCalendarNavigation.ts]: Consider merging with the main branch, this will (hopefully) raise linting errors

File: VLA-Frontend/src/hooks/useEventForm.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+export function useEventForm(initialDate?: string) {
+    // Main form fields (excluding recurrence fields).
+  const [formData, setFormData] = useState({
+    title: "",
+    category: "" as EventKind,
+    lectureId: "",
+    startDateTime: initialDate ? `${initialDate}T09:00` : "",
+    endDateTime: initialDate ? `${initialDate}T10:00` : "",
```
[the remark this answers is not in the document]
Comment (MarcoLeipnitz) [reply VLA-Frontend/src/hooks/useEventForm.ts]: In my eyes this is much more error prone, furthermore, why are there T09 and T10 hardcoded?

File: VLA-Frontend/src/hooks/useEventForm.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * useEventForm centralizes form state for EventForm.
+ * This hook can be used to keep EventForm component lean and focused on UI rendering.
+ */
+
+export function useEventForm(initialDate?: string) {
+    // Main form fields (excluding recurrence fields).
+  const [formData, setFormData] = useState({
+    title: "",
+    category: "" as EventKind,
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/hooks/useEventForm.ts]: This does not make sense for me, why can "" be considered as an EventKind when EventKind does not have the property ""

File: VLA-Frontend/src/hooks/useLectures.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import { useState } from "react";
+import type { Lecture } from "../components/calendar/CalendarTypes";
+
+/**
+ * useLectures stores and manages the list of lectures that can be assigned to events.
+ * Currently local-only state; later this can be replaced with backend persistence.
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/hooks/useLectures.ts]: Add a TODO here

File: VLA-Frontend/src/hooks/useEvents.ts [hunk 10]
```diff
@@ -0,0 +1,125 @@
+import { useState, useMemo } from "react";
+import type { CalendarEvent, Lecture } from "../components/calendar/CalendarTypes";
+import type { EventFormData } from "../components/calendar/EventForm/EventForm";
+import { addDays, toISODateLocal } from "../components/calendar/dateUtils";
+
+/**
+ * useEvents manages:
+ * - the list of calendar events (local state for now)
```
[the remark this answers is not in the document]
Comment (MarcoLeipnitz) [reply VLA-Frontend/src/hooks/useEvents.ts]: (I just saw line 21 but do that anyways, so that we can remove this when the backend is implemented properly)

File: VLA-Frontend/src/hooks/useEvents.ts [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+        title: formData.title,
+        dateISO: startDate,
+        calendarId: "user-events",
+        kind: formData.category,
+        status: formData.status,
+        subtitle: formData.people.join(", "),
+        startTime: startTime,
+        endTime: endTime,
+        lectureId: formData.lectureId,
+      });
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/hooks/useEvents.ts]: An event that is repeated exactly 0 times is a special case of your recurring events code, consider merging these two.
With that you would even save this surrounding if statement

File: VLA-Frontend/src/styles/CalendarView.css [hunk 12]
```diff
@@ -306,3 +309,388 @@
...
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background: rgba(0, 0, 0, 0.5);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  z-index: 1000;
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/styles/CalendarView.css]: We should establish a clear z-index hierarchy. As of now we are competing for the highest number without much consideration of the UI.

File: VLA-Frontend/src/styles/CalendarView.css [hunk 13]
```diff
@@ -306,3 +309,388 @@
...
+}
+
+/* ===== NEU: Add Button ===== */
+.cv-addBtn {
+  background: none;
+  border: 1px solid #d1d5db;
+  padding: 4px 12px;
+  border-radius: 6px;
+  font-size: 12px;
+  color: #6b7280;
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/styles/CalendarView.css]: This does not only apply to your new css components.
We should collect all of these colors into their respective :root variables as this seems to be getting a bit out of hand.

File: VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+};
+
+export type EventFormData = {
+  title: string;
+  lectureId?: string;
+  category: EventKind;
+  startDateTime: string;
+  endDateTime: string;
+  recurrence?: RecurrencePattern;
+  people: string[];
```
Comment (MarcoLeipnitz) [inline VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx]: I think we should make this a bit more general, so that we can support multiple input data types later

File: VLA-Frontend/src/components/calendar/DayColumn.tsx (hunk 3 above)
Comment (polodasi123) [reply VLA-Frontend/src/components/calendar/DayColumn.tsx]: Erledigt. Ich setze onClick nun nur noch bei vorhandenem onEventClick (via conditional spread). Nebenbei habe ich die Props gebündelt, um die JSX-Zeile lesbarer zu halten.

File: VLA-Frontend/src/components/calendar/EventForm/AddCategorySection.tsx [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+  categories,
+  selectedCategory,
+  onCategoryChange,
+  onAddCategory,
+}: AddCategorySectionProps) {
+  const [showAddForm, setShowAddForm] = useState(false);
+  const [newCategoryName, setNewCategoryName] = useState("");
+
+  const handleAdd = () => {
+    if (!newCategoryName.trim()) return;
```
[the remark this answers is not in the document]
Comment (polodasi123) [reply VLA-Frontend/src/components/calendar/EventForm/AddCategorySection.tsx]: done

File: VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx [hunk 16]
```diff
@@ -0,0 +1,224 @@
+import { useState } from "react";
+import type { EventKind, EventStatus , Lecture } from "../CalendarTypes";
+import AddLectureSection from "./AddLectureSection";
+import AddCategorySection from "./AddCategorySection";
+import RecurrenceSection from "./RecurrenceSection";
+
+export type RecurrencePattern = {
+  weekdays: number[];
```
[the remark this answers is not in the document]
Comment (polodasi123) [reply VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx]: Ich habe einen Union Type für Weekday erstellt (0-6)

File: VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+  endDate: string;
```
[the remark this answers is not in the document]
Comment (polodasi123) [reply VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx]: Ich bleibe bei String aus den gleichen Gründen wie bei startTime/endTime:

1. **Einfachheit**: endDate ist nur "yyyy-mm-dd" für die Anzeige
2. **Form-Kompatibilität**: HTML date inputs geben Strings zurück
3. **Keine Zeitzonen-Probleme**: Strings vermeiden Timezone-Komplexität
4. **Konsistenz**: Alle Datumsfelder sind Strings (dateISO, startDateTime, etc.)

Falls wir später Datum-Berechnungen brauchen, können wir zur Laufzeit 
Date-Objekte daraus erstellen. Für die Datenhaltung sind Strings die 
praktischere Wahl.

File: VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+      formData.recurrence = {
+        weekdays: recurrence.weekdays,
+        endDate: recurrence.endDate,
+      };
+    }
+
+    onSubmit(formData);
+  }
+  // Used to disable submit when required fields are missing
+  const isValid = title.trim() && category.trim() && startDateTime && endDateTime && endDateTime > startDateTime;
```
[the remark this answers is not in the document]
Comment (polodasi123) [reply VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx]: Zwischenvariablen verwendet, um es noch lesbarer zu machen.

File: VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+            >
+              Abbrechen
+            </button>
+            <button
+              type="submit"
+              className="cv-formBtn cv-formBtnSubmit"
+              disabled={!isValid}
+            >
+              Erstellen
+            </button>
```
[the remark this answers is not in the document]
Comment (polodasi123) [reply VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx]: kann man überlegen ob man das implemntiert

File: VLA-Frontend/src/styles/CalendarView.css (hunk 12 above)
Comment (polodasi123) [reply VLA-Frontend/src/styles/CalendarView.css]: siehe oben

File: VLA-Frontend/src/styles/CalendarView.css (hunk 13 above)
Comment (polodasi123) [reply VLA-Frontend/src/styles/CalendarView.css]: jemand mpsste sich mal ans css setzen, aber ich denke wir werden am ende eh nochmal ein ganz anderes design wählen, deswewgen das jetzt noch nicht so schlimm ist

File: VLA-Frontend/src/components/calendar/DayColumn.tsx [hunk 20]
```diff
@@ -1,22 +1,50 @@
...
+ * DayColumn represents exactly one day in the calendar.
+ * It receives the events for that day and renders them as clickable tiles.
  */
-export default function DayColumn({ day }: DayColumnProps) {
-  return (
-
-    //TODO: Events rendern
 
-     // data-date helps with debugging (DevTools) and later for tests
+function getStatusClass(status?: EventStatus): string {
```
[the remark this answers is not in the document]
Comment (MarcoLeipnitz) [reply VLA-Frontend/src/components/calendar/DayColumn.tsx]: getStatusStyleClass fände ich schöner, aber daran soll es nicht scheitern

File: VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx (hunk 19 above)
[the remark this answers is not in the document]
Comment (MarcoLeipnitz) [reply VLA-Frontend/src/components/calendar/EventForm/EventForm.tsx]: Schieb es einfach in ein Issue
